frontend/js/src/explore/fresh-releases/FreshReleases.tsx

```diff
@@ -217,29 +238,76 @@ export default function FreshReleases() {
         const releaseTags = Array.from(uniqueReleaseTagsSet);
         releaseTags.sort();
 
-        setReleases(cleanReleases);
-        setFilteredList(cleanReleases);
-        setAllFilters({
-          releaseTypes,
-          releaseTags,
-        });
-        setIsLoading(false);
+        return {
+          data: {
+            releases: cleanReleases,
+            releaseTypes,
+            releaseTags,
+          } as FreshReleasesData,
+          hasError: false,
+          errorMessage: "",
+        };
       } catch (error) {
-        toast.error(
-          <ToastMsg
-            title="Couldn't fetch fresh releases"
-            message={
-              typeof error === "object" ? error.message : error.toString()
-            }
-          />,
-          { toastId: "fetch-error" }
-        );
+        return {
+          data: {
+            releases: [],
+            releaseTypes: [],
+            releaseTags: [],
+          } as FreshReleasesData,
+          hasError: true,
+          errorMessage: error.message,
+        };
       }
-    };
-    // Call the async function defined above (useEffect can't return a Promise)
-    fetchReleases();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [pageType, range, showPastReleases, showFutureReleases, sort]);
+    },
+  });
+
+  const {
+    data: rawData = {
+      releases: [],
+      releaseTypes: [],
+      releaseTags: [],
+    } as FreshReleasesData,
+    hasError = false,
+    errorMessage = "",
+  } = loaderData || {};
+
+  const { releases, releaseTypes, releaseTags } = rawData;
+
+  React.useEffect(() => {
+    if (hasError) {
+      toast.error(
+        <ToastMsg
+          title="Couldn't fetch fresh releases"
+          message={errorMessage}
+        />,
+        { toastId: "fetch-error" }
+      );
+    }
+  }, [hasError, errorMessage]);
+
+  React.useEffect(() => {
+    if (releases.length > 0) {
+      setFilteredList(releases);
+    } else if (releases.length === 0 && filteredList.length > 0) {
```

Comment:
I ran into a weird warning.
```
Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.
```

The issue was that while fetching, the dependencies of this useEffect releaseTypes and releaseTags are empty arrays, and they were being replaced by empty arrays. I added a debugger log this change, and it says [] is being replaced by [].

This is why I added a special case to handle this change